contracts/VestingUpgradeable_V2.sol

```diff
@@ -0,0 +1,194 @@
+/// @dev This Smart Contract provides the vesting of tokens that will be unlocking until a certain date
+contract VestingUpgradeable_V2 is IVesting_V2, OwnableUpgradeable {
+    using SafeERC20Upgradeable for MyToken;
+    using Math for uint256;
+
+    /// @notice Store constant of percentage 100
+    /// @dev Store constant of percentage 100
+    uint256 public constant MAX_INITIAL_PERCENTAGE = 100;
+
+    /// @notice Store the initial timestamp in unix
+    /// @dev Store the initial timestamp in unix
+    uint256 public initialTimestamp;
+
+    /// @notice Store the last index of mapping of investors
+    /// @dev Store the last index of mapping of investors
+    uint256 public lastIndexInvestors;
+
+    /// @notice whether the investor was change
+    /// @dev whether the investor was change
+    bool public isInvestorChanged;
+
+    /// @notice Store the reward token
+    /// @dev Store the reward token
+    MyToken public rewardToken;
+
+    /// @notice Store the info about vesting by each type of allocation
+    /// @dev Store the info about vesting by each type of allocation
+    mapping(AllocationType => VestingInfo) public vestingInfo;
+
+    /// @notice Store the addresses of investors
+    /// @dev Store the addresses of investors
+    mapping(uint256 => address) public investors;
+
+    /// @notice Store amount of tokens given per investors
+    /// @dev Store amount of tokens given per investors
+    mapping(address => mapping(AllocationType => uint256)) public userTokens;
+
+    /// @notice Store amounts of reward tokens that were paid to recipients
+    /// @dev Store amounts of reward tokens that were paid to recipients
+    mapping(address => mapping(AllocationType => uint256)) public rewardsPaid;
+
+    /// @notice Initialize contract
+    /// @dev Initialize contract, sets reward token address & vesting info
+    /// @param token_ the reward token address
+    function initialize(address token_) external initializer {
+        require(token_ != address(0), "Incorrect token address");
+        VestingInfo memory info = VestingInfo(6 minutes, 10 minutes, 10);
+        vestingInfo[AllocationType.Seed] = info;
+        info.initialPercentage = 15;
+        vestingInfo[AllocationType.Private] = info;
+        rewardToken = MyToken(token_);
+
+        __Ownable_init();
+    }
+
+    /// @notice Move the amount of uncollected tokens from one investor address to another
+    /// @dev Move the amount of uncollected tokens from one investor address to another by owner
+    function changeInvestor() external virtual override onlyOwner {
+        require(!isInvestorChanged, "The investor has been already changed");
+        require(
+            userTokens[investors[0]][AllocationType.Seed] != 0,
+            "Investor has already taken away reward tokens"
+        );
+        isInvestorChanged = true;
+        userTokens[investors[1]][AllocationType.Seed] += userTokens[
```

Comment:
instead investors[0] & investors[1] use some realy eth address. and hardcode in code